node/libs/roles/src/attester/messages/msg.rs

```diff
@@ -0,0 +1,279 @@
+            !weighted_attester.is_empty(),
+            "Attester Committee must contain at least one attester"
+        );
+        Ok(Self {
+            vec: weighted_attester.values().cloned().collect(),
+            indexes: weighted_attester
+                .values()
+                .enumerate()
+                .map(|(i, v)| (v.key.clone(), i))
+                .collect(),
+            total_weight,
+        })
+    }
+
+    /// Iterates over weighted attesters.
+    pub fn iter(&self) -> impl Iterator<Item = &WeightedAttester> {
+        self.vec.iter()
+    }
+
+    /// Iterates over attester keys.
+    pub fn iter_keys(&self) -> impl Iterator<Item = &PublicKey> {
+        self.vec.iter().map(|v| &v.key)
+    }
+
+    /// Returns the number of attesters.
+    #[allow(clippy::len_without_is_empty)] // a valid `Committee` is always non-empty by construction
+    pub fn len(&self) -> usize {
+        self.vec.len()
+    }
+
+    /// Returns true if the given attester is in the attester committee.
+    pub fn contains(&self, attester: &PublicKey) -> bool {
+        self.indexes.contains_key(attester)
+    }
+
+    /// Get attester by its index in the committee.
+    pub fn get(&self, index: usize) -> Option<&WeightedAttester> {
+        self.vec.get(index)
+    }
+
+    /// Get the index of a attester in the committee.
+    pub fn index(&self, attester: &PublicKey) -> Option<usize> {
+        self.indexes.get(attester).copied()
+    }
+
+    /// Computes the leader for the given view.
+    pub fn view_leader(&self, view_number: ViewNumber) -> PublicKey {
+        let index = view_number.0 as usize % self.len();
+        self.get(index).unwrap().key.clone()
+    }
+
+    /// Signature weight threshold for this attester committee.
+    pub fn threshold(&self) -> u64 {
+        threshold(self.total_weight())
+    }
+
+    /// Compute the sum of signers weights.
+    pub fn weight(&self, signers: &Signers) -> u64 {
+        assert_eq!(self.vec.len(), signers.len());
+        self.vec
+            .iter()
+            .enumerate()
+            .filter(|(i, _)| signers.0[*i])
+            .map(|(_, v)| v.weight)
+            .sum()
+    }
+
+    /// Sum of all attesters weight in the committee
+    pub fn total_weight(&self) -> u64 {
+        self.total_weight
+    }
+
+    /// Maximal weight of faulty replicas allowed in this attester committee.
+    pub fn max_faulty_weight(&self) -> u64 {
+        max_faulty_weight(self.total_weight())
+    }
+}
+
+/// Calculate the consensus threshold, the minimum votes' weight for any consensus action to be valid,
+/// for a given committee total weight.
+pub fn threshold(total_weight: u64) -> u64 {
+    total_weight - max_faulty_weight(total_weight)
+}
+
+/// Calculate the maximum allowed weight for faulty replicas, for a given total weight.
+pub fn max_faulty_weight(total_weight: u64) -> u64 {
+    // Calculate the allowed maximum weight of faulty replicas. We want the following relationship to hold:
```

Comment:
add a comment that technically we need just n > 2*f, but for now we use a threshold consistent with the validator committee.

Reply:
Technically, we just need f+1.

Reply:
I would add said comment to the threshold() method above.

Reply:
I would add said comment to the `threshold` method above.

Reply:
in that case, let's just use validator::max_faulty_weight function instead for now.